ganttproject/src/net/sourceforge/ganttproject/client/RssFeedChecker.java

```diff
@@ -228,6 +236,14 @@ public void run() {
     };
   }
 
+  private void createUpdateDialog(String content) {
+    RssUpdate update = parser.parseUpdate(content);
+    if(update != null) {
+      Thread updateDialogThread = new Thread(() -> myUiFacade.showUpdateDialog(update));
```

Comment:
I don't think that creating Thread is necessary because UI must be shown in Swing thread and subsequent download must run NOT in Swing thread. 
`SwingUtilities.invokeLater` should do the trick.